### ISSUE TITLE
Symantec Protection identifies Process Hacker as a risk

### ISSUE THREAD

**Issue by tomasorti** [processhacker/processhacker#388]
Well, I know Process Hacker can not do anything with this...
But I wanted to inform about this issue.
Unfortunately the laptop where it is happening its from my company.
So for the moment I can not avoid Symantec to remove Process Hacker which is a hassle.

![symantec-ph1](https://user-images.githubusercontent.com/2485147/54220946-ccc8d900-44f2-11e9-959d-8f3f2015c3ad.png)
![symantec-ph2](https://user-images.githubusercontent.com/2485147/54220953-ce929c80-44f2-11e9-9da6-63071fe18121.png)

I also found this online:
https://www.symantec.com/security-center/writeup/2019-010717-0848-99?om_rssid=sr-mixed30days

**Comment by stdedos** [comment]
Can you somehow (ask to) update its signatures?

You can see also see related things at https://github.com/processhacker/processhacker/issues/356, but the virustotal signatures I have (just) checked, show that Symantec "Update 20190311" does not mark it as harmful.

**Comment by tomasorti** [comment]
This is what the supporting team is telling me:

> This detection is not false and the same will not be excluded as this is a risk for the environment.
> 
> Such tools like this mentioned by the user can be used for Injecting, Hooking to legit processes and this is a risk. Please advise the user to use another process tool.
> 
> please refer to the KB from Symantec. 
> 
> https://www.symantec.com/security-center/writeup/2019-010717-0848-99
> 

How can I prove them that Symantec "Update 20190311" does not mark Process Hacker as harmful. ?

**Comment by stdedos** [comment]
> This is what the supporting team is telling me

Support team of your "company environment"?
Tell them that Task Manager is also a potential security risk (sorry, couldn't resist the sarcasm).

> How can I prove them that Symantec "Update 20190311" does not mark Process Hacker as harmful. ?
* Show them the link? .. and
* Can you somehow (ask to) update its signatures?

**Comment by tomasorti** [comment]
> Support team of your "company environment"?
> Tell them that Task Manager is also a potential security risk (sorry, couldn't resist the sarcasm).

Ha, ha, ha, nice joke!
Yes, my company environment, indeed!

> > How can I prove them that Symantec "Update 20190311" does not mark Process Hacker as harmful. ?
> 
> * Show them the link? .. and
> * Can you somehow (ask to) update its signatures?

Which link? The [virustotal ](https://www.virustotal.com/el/file/9e0cf82ec6555fbdcf91c4dca9401098cc7a4081edea58cb8e0d878a770b04ee/analysis/1552421094/)one found on #356 ? I am confused...

**Comment by tomasorti** [comment]
Ok, I provided virustotal link.
Let's see what they say...

**Comment by zzColin** [comment]
TrendMicro OfficeScan that is required to be installed on my office computer reported the installer of PH as PUA.Win32.ProcHack.A about 6 days ago, and it reported the kprocesshacker driver as PUA.Win32.ProcHack.B today. Possibly this information would help.

(Sadly, OfficeScan in my corporate environment is technically prevented from being turned off or being uninstalled.)

**Comment by stdedos** [comment]
@zzColin I assume same as everyone else. If you, also, check the link above, the setup is being treated as trojan installer.

Hopefully @dmex can work his magic here too 😄. Hopefully not on a one-to-one basis.

**Comment by stdedos** [comment]
@zzColin I assume, the same as the OP. If you check the VirusTotal link, however, TrendMicro is treating the installer as `TROJ_GEN.R03FC0OC419` i.e. (IMHO) _Heuristically Trojan_. It is a rather more explicit classification than #356's (i.e. malware on the installer, clear executable); we don't know if OPs Symantec is catching the installer - VirusTotal is saying no.

However, if you check the definition of TrendMicro for PH is "really dedicated and explict": `PUA.Win32.ProcHack.A`:
* `PUA` -->  [PUP/PUA](https://en.wikipedia.org/wiki/Potentially_unwanted_program)
* `Win32` _for Windows_
* `ProcHack` --> PH
* And every part of it is assigned explicitly (i.e. `A` for the main executable, `B` for the driver)

_I also just noted that Symantec is also explicitly marking PH as a hacking tool_. Maybe Symantec (like Avast!), could agree that PH, without Terminator and the Kernel Driver it is "harmless", but no saying how or why PH was marked as a `PUA` on TrendMicro. PH is a _HackTool_ as "enough other programs", but `PUA`? It is a much more generic term.

----

@dmex this might "somehow" interest you: On the latest PH executable, someone (some-bot?) marked something regarding its detection status: https://www.virustotal.com/#/file/4ff1194547430bf84ea7bb9cec896e8bce1ab3dfdb0138e4942fa80ec91e2165/community

**Comment by tomasorti** [comment]
Hi there!
This is the response I got from the support team:

> Hello Team,
> 
> User can manually update SEP client by clicking on LiveUpdate button on the left on main SEP window. But this will not be necessary because client checks (by policy) for new updates once per 2 or 4 hours - depends on location - office/internet. Although his machine is already running with latest definitions. We can't trust information provided by third party products such as virustotal. Official statement from Symantec as vendor of SEP products was already provided by my colleague in previous update. This info was updated on 7th of Jan this year. At this stage we can't whitelist this product. 
> 
> Thanks.

Which is unfortunate...

**Comment by Dobatymo** [comment]
I was told by trend micro that processhacker was recently bundled with some malware. That's why it's detected now.

**Comment by stdedos** [comment]
Define `recently` and `some malware`.

`Terminator` and `kernel driver` have had bad mojo from multiple companies (e.g. anti-cheat, anti-virus), but it is not "malicious" per se (a metaphor would be that `Terminator` is like a really sharp knife - you can cook with it, or kill people).

Things also related to the security of the program: https://github.com/processhacker/processhacker/issues/268, https://github.com/processhacker/processhacker/issues/356#issuecomment-454740479

**Comment by Dobatymo** [comment]
@stdedos sorry, that's all the information I got. They didn't elabroate.

**Comment by tomasorti** [comment]
Hi again!
The IT team is pointing me to this link:

https://www.symantec.com/security-center/writeup/2019-010717-0848-99

Do you know if there is an update of that in Symatec web site?

**Comment by stdedos** [comment]
@tomasorti:
`such as injecting [...] processes`: that could be a fair assesment; however, @dmex could answer authoritatively on this.

`such as [...] ending processes`: So can Task Manager (hence my tantrum above https://github.com/processhacker/processhacker/issues/388#issuecomment-472324529). AFAIK, I don't think that PH can do "more" than Task Manager, on a given priviledge level, without the `Terminator` / `Kernel Driver` --> I don't think PH is worth the notoriety at this time.

Good luck convincing others though.

**Comment by zzColin** [comment]
New reports from TrendMicro OfficeScan today (definition version 14.909.71):
PE Viewer (peview.exe) -> `PUA.Win64.ProcHack.B.component`
main executable (ProcessHacker.exe) -> `PUA.Win64.ProcHack.C` and `PUA.Win32.ProcHack.C`

This effectively rendered PH totally unusable now, as our IT department sets OfficeScan to remove everything it detects without confirmation and deliberately locked its settings.

Sigh.

**Comment by stdedos** [comment]
It's my turn now :smiley:

BitDefender Endpoint Security Tools (Product versionL 6.69.134, Engines Version: 7.80239 [12798612]):
```
On-Access scanning has detected a threat. Access to the file has been denied.
C:\Users\user\AppData\Local\Temp\2\947422a5-7a9a-4a37-8223-716e2872301a.tmp is malware of type Application.ProcessHacker.1
```

However, I think in my case, it just wipes the kernel driver (_which, I would say, it's not a bad silver lining_). It still blocks automatic updates though :confused:

**Comment by tomasorti** [comment]
With definitions on Symantec updated to the following dates:

![image](https://user-images.githubusercontent.com/2485147/55474423-6165b900-5611-11e9-9e7a-32028321a00a.png)

Process Hacker is still considered a risk...

![image](https://user-images.githubusercontent.com/2485147/55474373-41ce9080-5611-11e9-8739-6853e5b0b677.png)

:disappointed:

**Comment by dmex** [comment]
> such as injecting [...] processes: that could be a fair assesment; however, @dmex could answer authoritatively on this.

That feature is blocked by the [ObRegisterCallbacks](https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/content/wdm/nf-wdm-obregistercallbacks) kernel API.

> such as [...] ending processes: So can Task Manager.

Termination is also blocked by the [ObRegisterCallbacks](https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/content/wdm/nf-wdm-obregistercallbacks) kernel API at low and medium (user) integrity.

The one and only exception is when the user is running with high integrity (System/Administrator) and calling [ZwTerminateProcess](https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/content/ntddk/nf-ntddk-zwterminateprocess) from kernel mode.

> Good luck convincing others though.

Stupid is as stupid does. Anyone can review the Process Hacker source-code here on Github and it's clearly not malicious. 

I talked with the MSRC (Microsoft Security Response Center) and they were not able to find any security issues with our source-code so why are these other companies claiming Process Hacker is malicious? 

![image](https://user-images.githubusercontent.com/1306177/55662747-c8a08a80-5816-11e9-8db4-12d270692527.png)

> New reports from TrendMicro OfficeScan today (definition version 14.909.71):

TrendMicro? They've were sending thousands of malformed requests to wj32.org between 2016-02-02 and 2018-05-23: https://i.imgur.com/pgsQSNo.png

> PE Viewer (peview.exe) -> PUA.Win64.ProcHack.B.component

How can anyone seriously consider peview.exe malicious? notepad.exe has more functionality than peview.exe does 😦

**Comment by stdedos** [comment]
> I talked with the MSRC (Microsoft Security Response Center) and they were not able to find any security issues with our source-code so why are these other companies claiming Process Hacker is malicious?

If only there was some standard way of "pleading your case" against the vendors and/or use that as some leverage at some point.

I am not familiar with any other vendor, but Avast! for the last x years doesn't not give you a way to "ignore" a detection. _Neither does it bother to explictly deteck PH, so that wouldn't be feasible in any level_.

**Comment by nanoant** [comment]
Same thing happened to me today. I use PH to monitor what is happening on my computer everyday. I fire up my machine today, and poof Symantec swallows my PH claiming it is [Hacktool.ProcHack!g1](https://www.symantec.com/security_response/detected_writeup.jsp?name=Hacktool.ProcHack%21g1&vid=4294908751).

This is really both hilarious and sad - It is like removing all knives from the cook's kitchen so the cook cannot hurt him/herself.

**Comment by nanoant** [comment]
FYI I just have built PH from sources, and it is working fine without alerting "the beast" 😉

**Comment by Coruscate5** [comment]
I'm basically in the same boat - our Sec team is a bit more knowledgeable so they are fine with me recompiling from source, but I can't seem to get the kernel driver to load properly. 

Whenever I run as admin (which is, of course, a separate account than the local user), PH complains that the kernel mode signature couldn't load (either the production kprocesshacker.sys, or one I signed myself) with: 

_**"Unable to verify the kernel driver signature."**_ 

Then either a detailed message saying "Object Name not Found" or "The cryptographic signature is invalid"

Any ideas/luck for anyone else? 

I've tried all sorts of signatures but I'm not a driver dev unfortunately, so I'm sure I'm missing something simple. I can open a new issue I just figured this group might've gotten through this already

**Comment by tomasorti** [comment]
> FYI I just have built PH from sources, and it is working fine without alerting "the beast" 😉

Really? If that is the case, I can consider building it.
I am currently pretty busy, but Easter is coming, so I will have time to do it.
I hope is not as time consuming as building [ccls](https://github.com/MaskRay/ccls), which builds Clang+LLVM, and takes too loooong...

**Comment by Coruscate5** [comment]
I'm not smart enough to quickly figure out why KPHVerifyClient fails the signature check, so I just disabled the check (since kprocesshacker.sys seems to be properly signed anyway by certs that I made, for both SHA1 and SHA256). 

Seems to be working fine for my purposes

**Comment by stdedos** [comment]
Well, probably the reason you fail detection, is because you have a different building environment _and_ you probably cannot sign it as "official" (as PH is not doing anything wrong, https://github.com/processhacker/processhacker/issues/388#issuecomment-480463694)

Not an expert in AV detection, but I guess PUPs are detected "hard-coded", not signature/heuristically-based

**Comment by nanoant** [comment]
@stdedos That is my impression too, and happily thanks to that I am still able to use PH to monitor my system, e.g. to see if there is something really suspicious going on or whatever garbage unless software is sucking my resources.

I am really amazed by the fact that PH is considered as a threat by different "authorities". Maybe because it has "hacker" in the name 🤦‍♂️

**Comment by tomasorti** [comment]
> I am really amazed by the fact that PH is considered as a threat by different "authorities". Maybe because it has "hacker" in the name 🤦‍♂️

That is what a friend of mine pointed... that Symantec could identify it as a risk because that...
If that is the case... It is terribly hilarious and stupid!

**Comment by stdedos** [comment]
_^^ see also https://github.com/processhacker/processhacker/issues/388#issuecomment-477989752_

**Comment by dmex** [comment]
> It is terribly hilarious and stupid!

You won't get reliable answers from those companies about why they're targeting Process Hacker - I've tried and not one vendor has given us an actual answer.... The hilarious thing about this is that literally nobody is buying their bullshit... Here's a graph based on Windows telemetry:

![image](https://user-images.githubusercontent.com/1306177/60769003-199c7500-a0cb-11e9-9bee-b392328a4b9d.png)

The red text shows when vendors have started targeting Process Hacker... I don't think anyone is buying into the bullshit from so called 'security' vendors claiming Process Hacker is malicious.

The project is open source and anyone can verify the source code and not one person has been able to show how Process Hacker is malicious... I've tried talking to a dozen security companies in the last year about Process Hacker and not even one company has been able to give me even the slightest information showing how Process Hacker could be abused or was malicious.

The MSRC (Microsoft Security Response Center) says they're no issues (https://i.imgur.com/TJnGws6.png) so I don't really care what some 3rd party vendor says (without at least some evidence which not one vendor has provided to date)...

The security industry is a complete joke and the things they're doing just prove they're a joke.

**Comment by Biswa96** [comment]
Just curious, where did you get the first graph?

**Comment by dmex** [comment]
> where did you get the first graph?

Windows telemetry for weekly execution statistics.

**Comment by tomasorti** [comment]
> FYI I just have built PH from sources, and it is working fine without alerting "the beast" 😉

By the way, about a month ago I tried to build from master and from [tag v2.39](https://github.com/processhacker/processhacker/releases/tag/v2.39) and it did not work well.
If I recall well, master did crash, and in v2.39 I do not see the Search text box.

Where should I built it from for a stable release with Search text box?

**Comment by dmex** [comment]
> Where should I built it from for a stable release with Search text box?

You need to build Plugins.sln

**Comment by dmex** [comment]
@stdedos 

> Well, probably the reason you fail detection, is because you have a different building environment and you probably cannot sign it as "official" Not an expert in AV detection, but I guess PUPs are detected "hard-coded", not signature/heuristically-based

AV uses hard-coded signatures from a few sources such as digital certificates, specific binary strings, specific PE header information and specific hashes of the binary or individual PE sections. Some vendors also have a 'reputation' signature which is determined based on the count of identical signatures seen on other customer machines and used to detect/block new or unkown signatures with a 'low' score - AV collects information about the software you're using or have installed and upload this telemetry information both for investigate purposes and 'reputation'.

You can use products such as vmprotect (https://vmpsoft.com/) and integrate it into the build process as a post-built event invoking the vmprotect commandline... A lot of commercial and open-source github projects are now using vmprotect for this or other reasons and it's something that we might use in the future.

**Comment by JakeSays** [comment]
Just thought I'd mention that Windows Defender now flags the nightly installer and zip file as trojans.

Here's what I'm seeing from the installer after downloading:

![image](https://user-images.githubusercontent.com/49178478/62348936-841eb600-b4bb-11e9-812c-61228096b1ca.png)

And from the zip file:
![image](https://user-images.githubusercontent.com/49178478/62349002-aa445600-b4bb-11e9-8e8c-ae203957f7d4.png)

**Comment by WiliTest** [comment]
> The MSRC (Microsoft Security Response Center) says they're no issues

actually windows 10 (1909) just flagged it as a threat now (maybe because I used today the feature "I/O priority" & "priority" to set an app on "high" to speed it ― though, I've no idea of the difference between both).

**Comment by dmex** [comment]
@WiliTest 

Its flagging the driver included with v2.39 but not the exact same driver included with the nightly builds: https://wj32.org/processhacker/nightly.php

**Comment by WiliTest** [comment]
@dmex windows (edge chromium) still doesn't like it: 

![](https://i.imgur.com/A9y3PSC.png)

**Comment by dmex** [comment]
@WiliTest 

Tell me something I dont know (like what the security issue actually is so I fix the code).

**Comment by nanoant** [comment]
> @WiliTest
> Tell me something I dont know (like what the security issue actually is so I fix the code).

@dmex Security issue is the application title. Being both unserious and serious. No point to get upset or fight the ppl that judge the app on its name. And honestly, what do you expect if you name the app Process Hacker? It’s like calling the company Kid Kidnappers and being surprised that police shows up at your door at some point.
I’d rename the app to sthing like Process Analyzer and put also clear statement on the website that app is not malicious which can be easily verified reading its sources.
If after that the app remains blacklisted, I’d send polite notice to blacklisters to include also SysInternals ProcExplorer since it poses same threat to the users.
Also, why this issue is closed if the issue still exist?

**Comment by stdedos** [comment]
> @dmex Security issue is in the application title. Being both unserious and serious. No point to get upset or fight the ppl that judge the app on its name. And honestly, what do you expect if you name the app Process Hacker? It’s like calling the company Kid Kidnappers and being surprised that police shows up at your door at some point.

If "a serious security application's" heuristics are based on a dictonary of bad words .... Well.

> If after that the app remains blacklisted, I’d send polite notice to blacklisters to include also SysInternals ProcExplorer since it poses same threat to the users.

He has already, and he is being one way or another ignored. Or, more explicitly _he claims he has done so and he is not given a clear response_ (https://github.com/processhacker/processhacker/issues/388#issuecomment-509000372), and I don't see much reason "not to believe him".

At the very least, Microsoft _has responded_ https://github.com/processhacker/processhacker/issues/388#issuecomment-480463694 and I would guess we will try to escalate it again for Edge Chromium.

> Also, why this issue is closed if the issue still exist?

Well ... there is always the debate: What to do with an issue that somebody **cannot possibly** solve, and depends on external factors that "don't come around".

I don't think that he wouldn't try to fix a known issue, if the issue has a somehow specific root cause and not vaguely defined (https://github.com/processhacker/processhacker/issues/388#issuecomment-558563256). He has also removed some functionality (Terminator and _something else_) to not be hated by _someone's anti-cheating mechanisms_ (Steam, Valve, ... someone else? I don't remember exactly)

**Comment by antoinebj** [comment]
IMHO, the reasons not to close the issue are:
1. The issue still exists.
2. To make it visible so as to avoid other people opening a similar issue.
3. To not give the initial impression to people who find this issue that there might be a workaround or solution and have to read through the replies hoping to find it.

I think this is a huge issue that can destroy the user base of this tool. I would think that if you want to make it live, you would try everything, including renaming the tool to avert bias from employees of antivirus editors. Don't think they're all geniuses. I also had apprehensions running something called Process Hacker, and recommending it to others.

**Comment by mrsshr** [comment]
I think it was flagged because the attacker killed the antivirus suite using the ProcessHacker command line + KProcessHacker. like volatility, nc.exe, etc

Process Hacker can kill any process (even protected) with:
```
ProcessHacker.exe -c -ctype process -cobject <PID> -caction terminate
```

**Comment by mrsshr** [comment]
Most enterprise users do not use Process Hacker.
It can also be seen as a risk in an enterprise environment.
If you don't need the functionality of KProcessHacker, use Procmon instead.

**Comment by speedwaystar** [comment]
Windows Virus & Threat Protection seem to have recently escalated their stance against Process Hacker. Today I got this:

!(https://i.imgur.com/THcpzd8.png)

**Comment by Aurakal** [comment]
Regarding Windows Defender, I can confirm this behavior here as well. Out of no where, on November 25th, it started quarantining Process Hacker 2 files and Registry items on my systems. And the HackTool:Win64/ProcHack page was last updated on November 25th on Microsoft's website.

https://www.microsoft.com/en-us/wdsi/threats/malware-encyclopedia-description?Name=HackTool:Win64/ProcHack&threatId=-2147221926

**Comment by dmex** [comment]
@nanoant 

A big part of being an adult is putting together information and discussing the facts but too often these days people are not using the facts to form an opinion and instead use their opinion as fact. It's hard to have an adult conversation with other people who have all the facts when you just contribute your uninformed opinion.

> He has already, and he is being one way or another ignored. Or, more explicitly, he claims he has done so and he is not given a clear response

You or anyone else can ask them for information and you won't get a clear answer either just buzz words with zero steps/instructions/source or other facts.


> what do you expect if you name the app Process Hacker? It’s like calling the company Kid Kidnappers and being surprised that police shows up at your door at some point.

The same legal protection as anyone else?
https://abr.business.gov.au/ABN/View?abn=44125908339

Process Hacker is the biggest competitor to Microsofts Process Explorer and the Windows Task Manager. 

Microsoft Defender is automatically removing a competitors product from Windows as "Malware" based on the information published here: ![image](https://user-images.githubusercontent.com/1306177/70043599-e24bd900-1614-11ea-9aea-6363f8efb10c.png)


The entire page is complete nonsense. Microsoft provided this response via the WDSI portal:

![image](https://user-images.githubusercontent.com/1306177/70044411-6fdbf880-1616-11ea-8fa8-329e4dcf6777.png)

Case closed. You can't reply to these tickets. Process Hacker is officially malware and will be automatically removed from Windows indefinitely as a high risk threat.

Microsoft have also never allowed third party task manager applications on Windows with changes to the Windows API hard-coding taskmgr.exe as the default Task Manager with more recent changes blocking third party task manager applications on Win10 from calling those same API functions.

Microsoft refuse to provide the development team with any information. So I can't make changes to the project that resolve the problem.

You will have to ask Microsoft why the project malware and publish their response somewhere.

**Comment by dmex** [comment]
Duplicate https://github.com/processhacker/processhacker/issues/454

**Comment by nanoant** [comment]
@dmex Take my apologies if I sounded offensive ors thing. You have my greatest respect for creating PH and for the actions you take to defend and keep it alive.

You are absolutely right that my comment was not based on the facts, but just on subjective experience. But the decisions taken in corporations are not always based on facts, and are made often by people that have their prejudices. My point there (which is not objective) was that "Process Hacker" name can serve them as an argument good enough (for them, and their executives) to put the app on the black-list. The reality might be more complicated though, e.g. if this is true that PH can kill Symantec relatively easily (or other AV engine), then this is something you don't want in corporate world.

> Microsoft refuse to provide the development team with any information. So I can't make changes to the project that resolve the problem.

Looking at the snapshots above, wow 🙄 Looks like great example of discriminative activities privileged - blocking legit software without providing reasonable explanation. And *"Hacktools can be use to patch or run software so it can run without valid license."* this is utter bul..it not an explanation. 😞 Maybe having open petition to stop Microsoft from unjustified discrimination and misusing their privileged position would help? I would sign that.

Just being me, I'd step back a bit, and maybe remove the intrusive features from the driver and ... yes, rename the app. Just to take all the arguments (even most ridiculous) from them. Unless there's some better idea what to do in such situation.

I still sustain my statement that issue remain open as long the problem exists, even if there is no viable solution at all. Even just to limit amount of duplicate issues, that will be open anyway as long there are some people that use PH and care about PH.

**Comment by stdedos** [comment]
> The reality might be more complicated though, e.g. if this is true that PH can kill Symantec relatively easily (or other AV engine), then this is something you don't want in corporate world.

For a fact, I cannot kill my antivirus engine (Avast!). That being said, maybe this is a cat-and-mouse game - i.e I need to "dismantle" the protection, I cannot just go full force at it.

All that being said, I'll paraphrase points raised here: Process Hacker is a collection of tools, code, and orchestration, in order to do "more than just" cute `taskmgr.exe`.
1. Knife is a "legal" tool too. That doesn't mean it is a good idea to throw it to people. Nor can you block people cutting their steak
2. A lot of what it does, it's based on having Administrative Priviledges. e.g., to run a prompt as `NT AUTHORITY\SYSTEM`, you should have `[...] permissions to install services` (if the escalation happens comparably to `psexec`). If you run as normal user and/or block loading the driver, then PH is far from the powerhouse it can be. _Note that **being** and Admin and **running** as an admin are two separate things since Win7._
3. _Disclaimer, I haven't read the source code, nor do I claim to have any knowledge of Windows APIs_. I don't think that PH is abusing the APIs. _Even if it did_, IMHO it's Microsoft's issue to patch vulnerabilities. If _that_ would be the issue, then they can freely study and patch their code.

Even though [3] would be "worth" blocking, I think it's bad behavior on their end regardless.

> I still sustain my statement that issue remain open as long the problem exists, even if there is no viable solution at all. Even just to limit amount of duplicate issues, that will be open anyway as long there are some people that use PH and care about PH.

It's called "pinning the issue", not "keeping the issue open" (unless only open issues can be pinned)

**Comment by dmex** [comment]
> For a fact, I cannot kill my antivirus engine (Avast!).

Avast does this by installing a hypervisor and hooking the SYSENTER/SYSEXIT syscalls for ntdll.dll exports. You could just as easily restart the process considering the antimalware driver doesn't stop executing when the usermode process terminates.

> My point there (which is not objective) was that "Process Hacker" name can serve them as an argument good enough (for them, and their executives) to put the app on the black-list.

You can setup policies with your own equipment. 

> PH can kill Symantec relatively easily (or other AV engine)

You can't disable and terminate an antivirus product by calling the process terminate function for a process and wrong assuming termination actually terminates an antivirus product.

The anti-malware driver and anti-malware services continue running and they restart the process after termination by registering for recovery using the Windows API or by using their own service and driver. 

Microsoft uses the same documented API for system components. Try terminate winlogon.exe and the process restarts while terminating wininit.exe bluescreens and restarts the machine while Antivirus products have similar functionality. 

> Just being me, I'd step back a bit, and maybe remove the intrusive features from the driver

This is not supported by facts. Process Hacker is a task manager for Windows using the process manager API for task managers with the exact same API functions as Microsoft's own software and tools since 2008 (11 years). You can go through [every file](https://github.com/processhacker/processhacker/tree/master/KProcessHacker) and all functions are documented on MSDN and used by other task manager software and Microsoft's own software.

**Comment by nanoant** [comment]
@dmex Okay then, you have valid arguments why PH cannot be used to dismantle AV and I will not trying to argue with them. All I have wrote were just speculations, why anyone would want to tear down PH. I think all what you wrote deserves to be put on the front page and README of this project. I think PH users including new potential ones that download it and find out that it is marked as a "virus", deserve to know.

I tried it to follow reasoning of [HackTool:Win64/ProcHack](https://www.microsoft.com/en-us/wdsi/threats/malware-encyclopedia-description?Name=HackTool:Win64/ProcHack&threatId=-2147221926) *"Hacktools can be used to patch or "crack" some software so it will run without a valid license or genuine product key."*, which can be attributed to any primitive debugger app, not to mention "sophisticated" tools like WinDbg. And this is only statement of why this is harmful. Rest is complete babble.

> Process Hacker is the biggest competitor to Microsofts Process Explorer and the Windows Task Manager.
> Microsoft Defender is automatically removing a competitors product from Windows as "Malware" based on the information published here: (...)

Competitors? In what sense? I had impression that you don't need to pay for SysInternals or TaskMan, hence don't see any financial benefit of MS' actions. Do you have some more evidence on this?

I could only (again) speculate (not based on facts, but on subjective experience) that PH exposes much more information of the internals of the Windows that any other tool, which may be sort of against way the latest computer product are sold to the consumers, who don't really own them, as they have absolutely no control and knowledge on what's happening on this hardware.

**Comment by WiliTest** [comment]
Just in case it could help: 
![2019-12-05_11-34-06](https://user-images.githubusercontent.com/24194840/70227682-4a083e00-1753-11ea-8c03-627a10e35fc4.png)

**Comment by speedwaystar** [comment]
i have no objection (in principle) to Windows Antivirus or Symantec or
whoever flagging powerful kernel-level tools which allows advanced users to
bypass system security as "potentially unwanted/dangerous," because for the
average user, they arguably are. many game trainers and cracks are
so-identified, which is fine, so long as the algorithm doesn't overstep the
bounds of reality and claim the software is actively malicious. falsely
identifying PH2 as a trojan, or a rootkit dropper, is quite another thing.
i completely understand that virus fingerprinting is a difficult and arcane
art that's hard to get reliably right, with polymorphicly evolving viruses
making it even more difficult, but Microsoft really should have a robust
mechanism in place to reverse false positives in cases like this.

**Comment by LiamKarlMitchell** [comment]
Well it's not just Microsoft, bit defender has been flagging it at work for several weeks...
Chrome too now...

How can I add it to the whitelist if I can't even install it?

**Comment by WiliTest** [comment]
@LiamKarlMitchell 

> "Check Quarantine threats and Restore as needed" 

Thanks, it works (the app wasn't uninstalled after all, but "just" quarantined) but, I guess I have already done that a week ago, and it has been quarantined again. (I'll confirm that if it happens again).

**Comment by dmex** [comment]
**Please use the correct topic:** https://github.com/processhacker/processhacker/issues/454

> Competitors? In what sense?

Task Manager, Process Explorer?

> I had impression that you don't need to pay for SysInternals or TaskMan, hence don't see clear financial benefit of MS' actions.

There are many financial and other benefits from eliminating your competition. 

One example would be if you consider how there has only been 4 or 5 updates to Task Manager in the last 19 years and how Process Hacker has been in constant development since 2009 with hundreds of updates (and includes so many features and statistics) that it will take significant funding and resources for Microsoft to include similar features in Task Manager. Microsoft probably determined it's a lot cheaper to just publish a webpage with false and misleading information and block anyone from being able to use Process Hacker than innovate and compete fairly by devoting money and resources to Task Manager development.

> PH exposes much more information of the internals of the Windows that any other tool

Microsoft could easily innovate and include that same information in their own products if they wanted. 

Everything you see included with Process Hacker is only possible because Microsoft makes that information or feature available for administrators and developers via the built-in Windows API and Process Hacker literally cannot show or do anything without those APIs.

You need as much information as possible these days to make informed decisions and determine when something is not secure, not setup correctly, malicious etc... More than a few companies have emailed us demands over the years that we remove XYZ features but then refuse to provide any facts that support their case.

> that may be sort of against corporate trend to sell hardware to the consumers, so they don't really own it and they have absolutely no control and/or knowledge on what's happening on that hardware.

This is how the entire antivirus industry operates. Antivirus products give zero control and/or knowledge about your machine and what anything on the machine is doing - these products are also not able to detect and terminate malicious software without regular ($$) updates.

Microsoft and other companies have spent millions on their products and they earn millions more from those constant updates - Process Hacker does not require constant updates for you to detect and terminate malicious software making those investments and their products practically meaningless so there is a huge incentive for Microsoft to protect that investment by labeling competitors products (Process Hacker) malicious and preventing users from being able to use it.

**Comment by LiamKarlMitchell** [comment]
It's actually used along side some trojans/viruses though. So it probably got caught up in that as it would make it a PUP for some?
https://www.crowdstrike.com/blog/doppelpaymer-ransomware-and-dridex-2/

**Comment by speedwaystar** [comment]
okay, this starts to make more sense.

In order to terminate some of these processes and services, DopplePaymer
uses an interesting technique that leverages ProcessHacker
<https://processhacker.sourceforge.io/>, a legitimate open-source
administrative utility. This application is bundled with a kernel driver
that can be used to terminate processes and services. DoppelPaymer is
bundled with six portable executable (PE) files that are encrypted and
compressed in the malware’s sdata section. These PE files contain 32-bit
and 64-bit versions of the following:

   - ProcessHacker application
   - ProcessHacker kernel driver
   - A custom stager DLL that is used to exploit ProcessHacker

**Comment by dmex** [comment]
@LiamKarlMitchell 
@speedwaystar 

> It's actually used along side some trojans/viruses though

I asked CrowdStrike months ago for copies of the samples but never heard back from them and I haven't been able to verify any information they've posted on that blog.

1) The hashes for the "_custom stager DLL that is used to exploit ProcessHacker_" don't seem to exist anywhere?

- 51d8618ec86159327e883615ad8989c7638172cf801f65ab0367e5b2e6af596a
- bfb7e62ba4ad5975e68a1beefb045cb72e056911fd7a8b070a15029dfcbbefe1

I haven't been able to locate these samples so do they actually exist?

2) The blog states "Once loaded, ProcessHacker’s kernel driver is leveraged to kill the blacklisted processes"

Windows does not allow processes to load kernel drivers unless you're running with elevated administrative privileges. This indicates the malware already had complete unrestricted access to the entire machine and was able to patch files/registry keys/literally do whatever without limitation and disable/overwrite/remove the KPH driver security checks.

The samples would help determine how it bypassed the KPH security checks but CrowdStrike hasn't provided the samples or anything else to the development team.

The kernel drivers are also still whitelisted by CrowdStrike:
https://www.hybrid-analysis.com/sample/70211a3f90376bbc61f49c22a63075d1d4ddd53f0aefa976216c46e6ba39a9f4?environmentId=110

![image](https://user-images.githubusercontent.com/1306177/70850094-9e8d8500-1eda-11ea-8ee9-d2c45c20f6ed.png)

I can't make any sense from it... If its true then why don't they share information with the development team so we can fix the issue instead of allowing attackers to keep abusing it?

Then there are some other questions like did they find the samples on clintons email server (to justify the secrecy) and how did the attackers breach the network and gain unrestricted administrative access but who knows.

What I do know is there are lots of accusations (by big companies) but scant substance backing it up

**Comment by antoinebj** [comment]
> One example would be if you consider how there has only been 4 or 5 updates to Task Manager in the last 19 years and how Process Hacker has been in constant development since 2009 with hundreds of updates (and includes so many features and statistics) that it will take significant funding and resources for Microsoft to include similar features in Task Manager. Microsoft probably determined it's a lot cheaper to just publish a webpage with false and misleading information and block anyone from being able to use Process Hacker than innovate and compete fairly by devoting money and resources to Task Manager development.

> Microsoft and other companies have spent millions on their products and they earn millions more from those constant updates - Process Hacker does not require constant updates for you to detect and terminate malicious software making those investments and their products practically meaningless so there is a huge incentive for Microsoft to try protect that investment by labeling Process Hacker malicious and preventing users from being able to use third party products.

Ok I'll be the devil's advocate.
There are several reasons why I think nobody needs go down that line of reasoning here.
1. This is not the Microsoft of the 90s, they have a different philosophy whereby they accept that not everything needs to be made or controlled by Microsoft: Linux is integrated into Windows, Edge uses Chromium, Microsoft apps on Android & iOS, 3rd-party open-source libraries in the development (.NET, VS, etc.) ecosystem, etc.
2. When they do try to crush competitors, it's by either buying them, competing on quality (don't laugh!), marketing, or by pre-installing in Windows, not by preventing the binaries from running. If they did, it would AFAIK be unprecedented.
3. This would be like a city trying to block a private but free recreational park for fear that its own public ones be abandoned when they're still well crowded.
4. I don't see how anybody at Microsoft would ever feel challenged by PH2, because even if nobody used Task Manager or Process Explorer anymore, and everybody used PH2 instead, then it would not change a single cent to their bottom line. Nothing that innovates on their ecosystem (Windows) without being a direct competitor to a product that they sell separately can hurt them. On the contrary, it's positive for them.
5. They're not the only ones blocking PH2.
6. They would have too much to lose in terms of public image if they were caught doing that. The risk is just not worth it.
7. Last but not least, the simplest explanation is almost always the most likely. Malicious software used PH2 (as described above) and the antivirus maintainers found it cheaper to simply use PH2's binaries as the signature, not caring at all about whether it's the right thing to do, or about the legitimate user-base that PH2 has. Simple as that.

**Comment by stdedos** [comment]
> 2. When they do try to crush competitors, [...] not by preventing the binaries from running. If they did, it would AFAIK be unprecedented.

What do you mean, exactly? Have you read (I don't remember where it is exactly written) that "Microsoft is allowed to remove programs without confirmation"?

https://www.howtogeek.com/243581/windows-10-may-delete-your-programs-without-asking/

Even though there *are* legit reasons, ... It is already "unprecedented" enough in my books

> 3. This would be like a city trying to block a private but free recreational park for fear that its own public ones be abandoned when they're still well crowded.

Yes, for the same reason EU decided to fine Microsoft for "forcefully" bundling IE and WMP.

> 5. They're not the only ones blocking PH2.

So, if "10 of us" would suddenly decide to bully someone, would that be different, because we have "legit reasons" and "market share" and "we are not nobody's in our field"? 😕 

----

I just picked some points to comment on. There could be much more.

The point that makes more sense, 7, _while it **does** make sense_ ... Wut?

So, if I would write a malware, that utilizes the `del /f` command ... does that mean that MS will treat `del` as a virus now?

----

> I can't make any sense from it... If its true then why don't they share information with the development team so we can fix the issue instead of allowing attackers to keep exploiting it?

If [1] is true, then the above statement should have been preposterous. Sadly, it seems that this is how everyone treats PH

> More than a few companies have emailed us demands over the years that we remove XYZ features but then refuse to provide any facts that support their case.

----

> Windows does not allow processes to load kernel drivers unless you're running with elevated administrative privileges. This would indicate the malware already had complete unrestricted access to the entire machine and was able to patch files/registry keys/literally do whatever without limitation and disable/overwrite/remove the KPH driver security checks.

I am suprised that the story does not end at "the attacker was given administrative priviledges". Even though UAC seems to treat some situations more leniently (I would think "non-important configuration options or 'secure' executables"?), it is end of story in my books if a malicious threat is escalated "not by its doing" (i.e. exploit bugs)

@dmex I am slightly suprised that the KPH driver needs to have some security checks. Does that mean security against exploitation, or e.g. that non-escalated user X cannot process owned by Y? (even though I don't understand why would X be able to load the KPH driver then)

**Comment by oldmud0** [comment]
I had a small thought: What if PH were a commercial product - would MS treat PH differently? One could argue that the competitor in this case (Process Hacker) is suffering economic loss from being "wrongfully" detected as malware, whereas some other similar product is not, and that Microsoft is intentionally marking it as malware to gain a competitive advantage in software debugging and power user tools. But as long as PH is FOSS, economic loss cannot be proven, so MS can safely win a case.

So I wonder if Microsoft's attitude would suddenly change if money started moving - like if there existed a commercial support license for PH's kernel driver. Then a party representing PH could claim economic loss.

At any rate, this sounds like a legal issue, and it doesn't sound like anything else but large-scale petitioning would stop MS from blocking open-source tools that just happen to be robust enough to have malicious uses.

**Comment by WiliTest** [comment]
> robust enough to have malicious uses.
That one looks weird.

**Comment by dmex** [comment]
> This is not the Microsoft of the 90s, they have a different philosophy whereby they accept that not everything needs to be made or controlled by Microsoft: Linux is integrated into Windows, Edge uses Chromium, Microsoft apps on Android & iOS, 3rd-party open-source libraries in the development

Yes, this is true for Linux and markets like browsers but not at all when it comes to the task manager market. 

It's a fact that Microsoft does not document the API required by competitors to compete fairly with Microsoft's Task Manager or Process Explorer and has changed other API functions to block third party software including having hard-coded taskmgr.exe throughout the entire Windows operating system.

Task Manager and Process Explorer have features that we still can't implement in Process Hacker because they're not documented and we haven't been able to figure out how it was done.

Microsoft has also never provided any information or documentation to the development team or followed up on any issues we've reported including major issues such as LTSB versions of Windows blue screening under some scenarios when running Process Hacker (caused by changes to the Windows API that only affect Process Hacker) or literally any other issue we've found or encountered on Windows.

> When they do try to crush competitors, it's by either buying them, competing on quality (don't laugh!), marketing, or by pre-installing in Windows, not by preventing the binaries from running. If they did, it would AFAIK be unprecedented.

How would you describe the reasoning and actions by Microsoft to classify Process Hacker as malicious software, block downloads from our website and automatically remove the product from users machines?

https://i.imgur.com/wRAeASJ.png
https://i.imgur.com/mO1DdRE.png

> They're not the only ones blocking PH2.

There is a major legal difference between Microsoft and third party companies.

> They would have too much to lose in terms of public image if they were caught doing that.

I don't understand it either.

> Malicious software used PH2 (as described above) and the antivirus maintainers found it cheaper to simply use PH2's binaries as the signature, not caring at all about whether it's the right thing to do, or about the legitimate user-base that PH2 has. Simple as that.

That might be true but its not legal (except maybe China). I've read just about everything published online and the only thing that I have seen is:

1) attacker exploits or bruteforces the Remote Desktop Protocol (RDP) owned by Microsoft.
2) attacker logs into the administrative desktop via RDP.
3) attacker executes programs on the desktop with administrative privileges - Chrome, Command Prompt, Powershell, Process Hacker...

So why are companies blaming Process Hacker when attackers have administrative access and can uninstall, disable, remove, change anything on the machine? It's because they're not covered by insurance for their own incompetence.

The staff won't admit to making a massive mistake by allowing RDP connections from the internet and not configuring any local security policies needed to make it secure (lockout, complexity etc) so it's more convenient to blame the programs the attacker executed.

> I am suprised that the story does not end at "the attacker was given administrative priviledges".

Microsoft and others won't do anything when an attack requires administrative privileges so they're determined to leave out that fact.

https://devblogs.microsoft.com/oldnewthing/?p=15273
https://devblogs.microsoft.com/oldnewthing/?p=94505
https://devblogs.microsoft.com/oldnewthing/?p=3883

> as long as PH is FOSS, economic loss cannot be proven, so MS can safely win a case.

No. Competition laws are not based on monetary losses and include anything that causes any sort of harm to consumers and/or competitors - the laws are especially severe and restrictive when the company is a declared monopoly.

I would think it's obvious how consumers (users of Process Hacker) have been harmed by Microsoft's actions. I've personally had a contract with Electronic Arts practically cancelled by Microsoft's actions. The project has also been harmed by Microsoft refusing to share interoperability information and incompatible design changes to Windows excluding task manager products and by hard-coding taskmgr.exe throughout Windows. Process Hacker is also not the only company affected with the SANS institute and Winsider Seminars & Solutions both have training courses with Process Hacker and face increased costs from helping students reconfigure their machines etc... and we're not even getting into libel and other conduct.

> I am slightly suprised that the KPH driver needs to have some security checks. Does that mean security against exploitation, or e.g. that non-escalated user X cannot process owned by Y? (even though I don't understand why would X be able to load the KPH driver then)

Multiple users can be logged onto the machine simultaneously so our kernel driver enforces a privilege check to prevent non-administrators from preforming administrative actions they're not authorized to preform (such as terminating processes owned by other users).

**Comment by oldmud0** [comment]
> No. Competition laws are not based on monetary losses and include anything that causes any sort of harm to consumers and/or competitors - the laws are especially severe and restrictive when the company is a declared monopoly.

Sure, not per se, but it's much easier to prove the case if you can claim economic loss. A FOSS project can't do that by itself - in this case, PH users did not transact for any promise to be able to use the software (and even if there was a transaction, the GPL explicitly waives the warranty of merchantability), and PH does not materially benefit regardless of how many users are able to run the software. Heck, the project doesn't even take donations, so the maintainers cannot claim that a loss of users caused a decrease in donations.

The bottom line is that the PH project by itself can't make the case - another party who has materially lost something as a result of Microsoft's reckless actions (such as the ones you mentioned) would have to make the case.

**Comment by dmex** [comment]
> Heck, the project doesn't even take donations

There is a donation link in the main menu > About > Donate:
![image](https://user-images.githubusercontent.com/1306177/70927685-8dcc4300-2083-11ea-9eee-c3773a6adf92.png)

There's also a donation link on download page on the project website? https://processhacker.sourceforge.io/downloads.php

> The bottom line is that the PH project by itself can't make the case - another party who has materially lost something

This is now how it works (especially outside the USA like we are).

**Comment by oldmud0** [comment]
Whoops, didn't see those.

I will admit that I am not familiar with law outside of the US - specifically how to establish grounds for a case. In the US, generally a party has to prove that they were harmed by someone's wrongful actions. 

So, let me summarize. PH's argument is "our users can't run our software because Microsoft is blocking it, and this violates an anticompetitive law because Microsoft has a software just like it that it allows. Therefore, we demand Microsoft to allow our software to run, so long as Microsoft has a similar product." Is this correct? I still don't think this would be an easy case to argue from PH's side.

**Comment by AndrewDP23** [comment]
TrendMicro just picked this up on my enterprise workstation. Process Monitor has been there for a long time, but my workstation just received a TrendMicro update that identifies PUA.Win64.ProcHack.B.component and PUA.Win32.ProcHack.C from Process Monitor.

I wouldn't be surprised if our security team ask me (with a concerned look) why I have something called Process *Hacker* on my workstation, which their authoritative source of threat detection picks up as a hacking tool.

What a disappointment from Trend. Process Monitor does well, and I hope this interference can be overcome - it is a valuable tool, and your great work is much appreciated.

**Comment by LiamKarlMitchell** [comment]
A strange question, but has anyone started a fork for a Lite build of Process Hacker with things removed and strings changed as to not arouse suspicion by w/e these anti viruses are complaining about?

Its a great tool even with reduced functionality to just use a task manager and resource monitor.
Maybe released as a new name such as "Better Process Manager" removing the "hacker" functionalities as well.

One that would work in business environments where we are unable/unwiling to flag it as allowed.

I guess I can try Process Explorer thanks :) (I've used that one in the past heh)

**Comment by dmex** [comment]
>  if our security team ask me (with a concerned look) why

If they have to ask then you'll know they're incompetent and can't do anything on their own ;P

> What a disappointment from Trend.

Trend Micro been sending thousands of malformed http requests to our updater server @ wj32.org for more than two years and always the same static IP address registered to their office in Japan.

> I hope this interference can be overcome

They've never once replied to support tickets or emails. I don't know what their problem would be.

> has anyone started a fork for a Lite build

There are 576 forks on github that I can see from the Github forks page and even they're being targeted regardless of what parts they're using. For example: https://github.com/lucasg/Dependencies/issues/26

> not arouse suspicion by w/e these anti viruses are complaining about?

I'm considering integrating VMProtect from https://vmpsoft.com/ since it encrypts the binary and dynamically changes code execution rendering anti-virus signatures obsolete - anyone can use it themselves too since it only requires the binary and not the source-code.

> Maybe released as a new name such as "Better Process Manager" removing the "hacker" functionalities as well.

There is no "hacker functionality"? We use the exact same API functions Microsoft does in their own products and the name was based on the dictionary definition - having to reverse engineer Windows to compete with Task Manager and Process Explorer.

> I guess I can try Process Explorer thanks :) (I've used that one in the past heh)

This is the entire point isn't it? Force everyone away to competing products so they're not able to protect themselves without an antivirus product?

**Comment by stdedos** [comment]
> > not arouse suspicion by w/e these anti viruses are complaining about?
> 
> I'm considering integrating VMProtect from [vmpsoft.com](https://vmpsoft.com/) since it encrypts the binary and dynamically changes code execution rendering anti-virus signatures obsolete - anyone can use it themselves too since it only requires the binary and not the source-code.

Can I has it? Can I has it? Can I has it? Can I has it? Can I has it? :smile:

**Comment by dmex** [comment]
@stdedos 

VMProtect can be used right now. You just need to install it and select the executable then it'll produce a protected executable that can be distributed anywhere.

**Comment by stdedos** [comment]
> > What a disappointment from Trend.
> 
> Trend Micro been sending thousands of malformed http requests to our updater server @ wj32.org for more than two years and always the same static IP address registered to their office in Japan.

I've been wondering though, why do you bother with their traffic though, and not block them (expecially with these patterns).

**Comment by dmex** [comment]
> I've been wondering though, why do you bother with their traffic though, and not block them (expecially with these patterns).

It would be easily twisted into justifying their actions towards the project.

**Comment by mickae1** [comment]
same problem with kaspersky !

**Comment by bitraid** [comment]
it is amusing to see that some antivirus [detect a threat](https://www.virustotal.com/gui/file/a0c005047e97f2acb0368d4fa9838f3eef1ce60fc1f1d46f4992bb4a72ecc84d/detection) in a program that just calls `ExitProcess` (compile with fasm):
```asm
format PE GUI
entry start

section '.text' code readable executable

  start:
    call    [ExitProcess]

section '.idata' import data readable writeable

  dd 0,0,0,RVA kernel_name,RVA kernel_table
  dd 0,0,0,0,0

  kernel_table:
    ExitProcess dd RVA _ExitProcess
    dd 0

  kernel_name db 'KERNEL32.DLL',0

  _ExitProcess dw 0
    db 'ExitProcess',0
```

**Comment by dmex** [comment]
> same problem with kaspersky !

Kaspersky was the company caught using their antivirus to steal documents... why does anyone continue using it?

Try something like Malwarebytes instead since they don't block tools like Process Hacker for showing system activity:
https://www.malwarebytes.com/premium/

> it is amusing to see that some antivirus detect a threat in a program that just calls ExitProcess

It's not the code but rather the binary hash and timestamp. Some companies just consider all binaries with recent timestamps malicious to fool unsuspecting customers into thinking those products protected them from threats which don't exist.

Most antivirus products should be sued for their behaviour but section 230 of the Communications Decency Act (CDA) gives antivirus software immunity for otherwise illegal behaviour.

**Comment by mickae1** [comment]
> > same problem with kaspersky !
> 
> Kaspersky was the company caught using their antivirus to steal documents... why does anyone continue using it?
> 
> Try something like Malwarebytes instead since they don't block tools like Process Hacker for showing system activity:
> https://www.malwarebytes.com/premium/
> 
> > it is amusing to see that some antivirus detect a threat in a program that just calls ExitProcess
> 
> It's not the code but rather the binary hash and timestamp. Some companies just consider all binaries with recent timestamps malicious to fool unsuspecting customers into thinking those products protected them from threats which don't exist.
> 
> Most antivirus products should be sued for their behaviour but section 230 of the Communications Decency Act (CDA) gives antivirus software immunity for otherwise illegal behaviour.

I don't know, but my company is using kaspersky :( and I can't use processhacker :'(
